packages/sms-reminders/helper/get-appointment.ts

```diff
@@ -1,24 +1,164 @@
+	pn2.given_name  as NOK_FIRST_NAME,
+	pn2.middle_name as NOK_MIDDLE_NAME,
+	pn2.family_name as NOK_LAST_NAME,
+	convert(p2.birthdate, char) as NOK_DATE_OF_BIRTH,
+	case when pa2.person_attribute_type_id = 25 then pa2.value end as NOK_PHONE_NUMBER,
+	p2.gender as NOK_GENDER,
+	p2a.address1 as NOK_ADDRESS,
+	max(case when pi2.identifier_type = 8 then pi2.identifier end) as GODS_NUMBER,
+	max(case when (pi2.identifier_type = 28 or pi2.identifier_type = 29)  then pi2.identifier end) as IPI_IDENTIFIER_TYPE_1_ID,
+	max(case when (pi2.identifier_type = 28 or pi2.identifier_type = 29)  then pit.name end) as IPI_IDENTIFIER_TYPE_1_NAME,
+	max(case when pi2.identifier_type = 45 then pi2.identifier end)as IPI_IDENTIFIER_TYPE_2_ID,
+	max(case when pi2.identifier_type = 45 then pit.name  end)as IPI_IDENTIFIER_TYPE_2_NAME,
+	max(case when r.relationship  = 7 then "married" end) as MARITAL_STATUS,
+	convert(fh.weight, char) as WEIGHT,
+	convert(fh.height, char) as HEIGHT,
+	convert(date(fh.encounter_datetime), char) as VISIT_DATE,
+	et.name as PATIENT_TYPE,
+	convert(date(fh.enrollment_date), char) as HIV_CARE_ENROLLMENT_DATE,
+	convert(date(fh.rtc_date), char) as APPOINTMENT_DATE,
+	u.username as PLACER_ENTITY,
+	e.creator  as PLACER_NUMBER
+from amrs_migration.person p 
+left join amrs_migration.person_name pn on p.person_id = pn.person_id and pn.voided = 0
+left join amrs_migration.person_address pa on p.person_id = pa.person_id and pa.voided = 0
+left join amrs_migration.patient_identifier pi2 on p.person_id = pi2.patient_id and pi2.voided = 0
+left join amrs_migration.patient_identifier_type pit on pi2.identifier_type = pit.patient_identifier_type_id
+left join amrs_migration.person_attribute pa2 on p.person_id = pa2.person_id and pa2.voided = 0
+left join amrs_migration.person_attribute_type pat on pa2.person_attribute_type_id = pat.person_attribute_type_id
+left join amrs_migration.relationship r on p.person_id  = r.person_a and r.voided = 0  
+left join amrs_migration.relationship_type rt on r.relationship = rt.relationship_type_id
+left join ndwr.flat_vitals fv on p.person_id = fv.person_id 
+left join etl.flat_hiv_summary_v15b fh on fh.person_id = p.person_id and fh.is_clinical_encounter = 1 and fh.next_clinical_datetime_hiv is null
+left join amrs_migration.person p2 on p2.person_id = r.person_b and p2.voided = 0
+left join amrs_migration.person_name pn2 on p2.person_id = pn2.person_id and pn2.voided = 0
+left join amrs_migration.person_attribute pa3  on pa3.person_id = p2.person_id and pa3.voided = 0
+left join amrs_migration.person_address p2a on p2a.person_id = p2.person_id  and p2a.voided = 0
+left join amrs_migration.encounter_type et on fh.encounter_type = et.encounter_type_id 
+left join amrs_migration.encounter e on e.encounter_id = et.encounter_type_id 
+left join amrs_migration.users u on u.user_id = e.creator 
+inner join ndwr.mfl_codes mc on pi2.location_id  = mc.location_id 
```

Comment:
Why do we have all these joins? Some are even duplicated.

Reply:
Refactoring it.